src/functions.js

```diff
@@ -36,43 +42,85 @@ function cities( capitals, formatter ){
   // such as 'Paris is the capital of France'
 
   // apply formatter to each object in capitals array and return an array of resulting sentences
+  //var capI = capitals[i];//console.log('capital[i] is', capI);//var item = capI[key];
+  //console.log('item is', item, 'key is', key);
+  /* //second approach
+ var a=[]
+  for (var i = 0; i < capitals.length; i++) {
+      a.push(formatter());
+  }
+  return a;
+  function formatter (city, country) {
+   return `${capitals[i]['city']} is the capital of ${capitals[i]['country']}`;
+  } */
+/* //map approach mine weird but still works!
+  var a=[]
+  for (var i = 0; i < capitals.length; i++) {
+      a.push(formatter(capitals[i]));
+  }
+  return a;
+  var formatter= 
+    capitals.map ((city, country)=> 
+   `${capitals[i]['city']} is the capital of ${capitals[i]['country']}`);
+}  */
+//best Lor solution i was waiting to see it!
+var resultArray = capitals.map(arrayElement => `${arrayElement.city} is the capital of ${arrayElement.country}`);
+return (resultArray);
 }
 
+
+
 function largerThanTen( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only numbers from the input array which are greater than 10
   // hint: filter
+  return numbers.filter(i=> i>10);
 }
 
 function even( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only even numbers from the input array
   // hint: you may want to use the modulus operator '%'
+  //numbers.filter(i => (i%2 ==0)?i:0 );
+  //for(var i=0;i<numbers.length;i++){
+   return numbers.filter(i => !(i&1))
 }
```

Comment:
Whilst the solution works fine and passes the test (as do all the others), this one could be written differently to make it more readable/understandable at a later date. Think of code as something humans need to be able to read, rather than the machine, as it makes supporting solutions much easier in the future.